### DISCUSSION TITLE
What subgroup operation does MoltenVK support?

### DISCUSSION THREAD

**Discussion by FunMiles** [KhronosGroup/MoltenVK#1529] [General]
I have been looking into subgroup operations in support of Vulkan compute.
One of the basic tools of GPU Compute codes is a prefix sum. I tried to use an interesting implementation with a shader at https://github.com/linebender/piet-gpu/blob/prefix/piet-gpu-hal/examples/shader/prefix.comp 
Unfortunately, glslangValidator installed from LunarG gives me the following error messages when compiling:
```
ERROR: /Users/michel/CLionProjects/GPUGraph/shaders/prefix.comp:59: 'subgroup op' : requires SPIR-V 1.3 
ERROR: /Users/michel/CLionProjects/GPUGraph/shaders/prefix.comp:59: '' : compilation terminated 
ERROR: 2 compilation errors.  No code generated.
```
The error is with `subgroupInclusiveAdd` and the same error appears where `subgroupBroadcast` is called.

Does that mean I am out of luck or is there a way to get those function to work with MoltenVK?

**Comment by FunMiles** [comment]
Answering my own discussion. Simply adding a flag (`--target-env vulkan1.1 `) to the call to glslangValidator call did the trick. So it seems all the subgroup operations I rely on do work.

**Comment by cdavis5e** [comment]
Yeah, you need to enable Vulkan 1.1 to enable subgroup operations. Since you're compiling your shaders from GLSL source, as you discovered, you need to do this on the GLSL compiler as well.
